tools/slack-release-notifier/src/main.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * @license
+ * Copyright 2019 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getInput } from '@actions/core';
+import { context, GitHub } from '@actions/github';
+import { IncomingWebhook } from '@slack/webhook';
+import { ReposGetReleaseByTagResponse } from '@octokit/rest';
+
+import { getReleaseType, getTextForReleaseType } from './utils/release-type';
+import { convertToMrkdwn } from './utils/convert-to-mrkdwn';
+import { findPreviousRelease } from './utils/find-previous-release';
+
+function getReleaseTitle(release: ReposGetReleaseByTagResponse) {
+  if (!release.name) {
+    return release.tag_name;
+  }
+
+  return release.name.includes(release.tag_name)
+    ? release.name
+    : `${release.name} (${release.tag_name})`;
+}
+
+async function run(): Promise<void> {
+  const slackWebhookUrl = process.env.SLACK_WEBHOOK_URL;
+  const githubToken = getInput('repo-token', { required: true });
+
+  if (!slackWebhookUrl)
+    throw new Error('Required environment variables are not set.');
+
+  if (!context.ref.includes('/')) {
+    throw new Error('Malformed tag ref');
+  }
+
+  // The ref in the context is formatted like 'ref/tags/1.0.0'.
+  // The API expects only the tag string e.g. 1.0.0
+  const splitTag = context.ref.split('/');
+  const tag = splitTag[splitTag.length - 1];
+
+  const client = new GitHub(githubToken);
+
+  const release = (
+    await client.repos.getReleaseByTag({
+      owner: context.repo.owner,
+      repo: context.repo.repo,
+      tag: tag,
+    })
+  ).data;
+
+  // To find out if we should announce a major, minor or patch release,
+  // we need to compare with the previous release version.
+  let previousRelease = await findPreviousRelease(client, tag);
```

Comment:
We already did these checks before the release - at that point the version is already released. If it the patch version is greater than 0 we have a patch version if the minor version is greater than 0 we have a minor version and otherwise we have a major version. Special cases are rc versions but they can be easily identified by the rc at the end :)

Reply:
You can use the semver package for that :) We already have it in the library - you can search for it